broker.go

```diff
@@ -1293,71 +1292,92 @@ consumeRetryLoop:
 			continue
 		}
 
-		for _, topic := range resp.Topics {
-			if topic.Name != c.conf.Topic {
-				c.conf.Logger.Warn("unexpected topic information received",
-					"got", topic.Name,
-					"expected", c.conf.Topic)
+		messages, shouldRetry, err := extractMessages(resp, c.conf)
```

Comment:
I factored this out so that I could test it without invoking the entire machinery of a server and network connections and all that.  Our primitive server implementation doesn't produce record batches, so that would have been a whole lot more work to add.